src/dune_rpc_server/dune_rpc_server.ml

```diff
@@ -481,29 +481,48 @@ module H = struct
           | Cancelled
       end
 
-      let on_cancel map _session poller =
+      let fill_if_possible ivar value =
+        let* current = Fiber.Ivar.peek ivar in
+        match current with
+        | Some _ -> Fiber.return ()
+        | None -> Fiber.Ivar.fill ivar value
+
+      let on_cancel map pending _session poller =
         let new_map =
           Map.update !map poller ~f:(function
             | None -> assert false
             | Some Status.Cancelled as s -> s
             | Some (Active _) -> Some Cancelled)
         in
         map := new_map;
-        Fiber.return ()
+        match !pending with
+        | None -> Fiber.return ()
```

Comment:
This case occurs when cancellation occurs while there's no request pending, right?

Reply:
Yes, that's the idea.

Reply:
Okay. A comment would be useful here.